homeassistant/components/fitbit/sensor.py

```diff
@@ -135,6 +135,17 @@ def _water_unit(unit_system: FitbitUnitSystem) -> UnitOfVolume:
     return UnitOfVolume.MILLILITERS
 
 
+def _int_value_or_none(field: str) -> Callable[[dict[str, Any]], int | None]:
+    """Value function that will parse the specified field if present."""
+
+    def convert(result: dict[str, Any]) -> int | None:
+        if value := result["value"].get(field):
```

Comment:
If the value is a number this could be falsy for zeros.

Reply:
Changed the behavior to explicitly test for `None` and extended the tests to have a few variations.